### PR TITLE
PreviousGainOrLoss page - controller, view, model, form and associated specs.

### PR DESCRIPTION
**Pull request by euptonqa**

### CODE REVIEW

File: app/views/calculation/nonresident/previousLossOrGain.scala.html [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+@(previousLossOrGainForm: Form[PreviousLossOrGainModel])(implicit request: Request[_])
+
+@main_template(Messages("calc.previousLossOrGain.question")) {
+
+    <a id="back-link" class="back-link" href="@controllers.nonresident.routes.OtherPropertiesController.otherProperties()">@Messages("calc.base.back")</a>
+
+    @errorSummary(previousLossOrGainForm, "previousLossOrGain")
+
+    <h1 class="heading-xlarge">@Messages("calc.previousLossOrGain.question")</h1>
+
```
Comment (TomDudley92) [inline app/views/calculation/nonresident/previousLossOrGain.scala.html]: Missing the side panel from the prototype.

File: test/views/nonResident/PreviousLossOrGainViewSpec.scala [hunk 2]
```diff
@@ -115,6 +115,51 @@ class PreviousLossOrGainViewSpec extends UnitSpec with WithFakeApplication with
...
+        }
+
+        "has a link with the class 'external-link'" in {
+          linkOne.attr("class") shouldBe "external-link"
+        }
+
+        "has a link with a rel of 'external'" in {
+          linkOne.attr("rel") shouldBe "external"
+        }
+
```
Comment (TomDudley92) [inline test/views/nonResident/PreviousLossOrGainViewSpec.scala]: missing a test for href

File: test/forms/nonResident/PreviousLossOrGainFormSpec.scala [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+ */
+
+package forms.nonResident
+
+import uk.gov.hmrc.play.test.{UnitSpec, WithFakeApplication}
+import models.nonresident.PreviousLossOrGainModel
+import forms.nonresident.PreviousLossOrGainForm._
+import assets.MessageLookup.NonResident.{PreviousLossOrGain => messages}
+
+/**
```
Comment (miranda-hawkes) [inline test/forms/nonResident/PreviousLossOrGainFormSpec.scala]: Can you remove this comment please?

File: test/forms/nonResident/PreviousLossOrGainFormSpec.scala [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        form.errors.length shouldBe 1
+      }
+
+      s"contain the error message ${messages.mandatoryCheck}" in {
+        form.errors.head.message shouldBe messages.mandatoryCheck
+      }
+    }
+
+    "an invalid string is provided" should {
+      lazy val map = Map(("previousLossOrGain", ""))
```
Comment (miranda-hawkes) [inline test/forms/nonResident/PreviousLossOrGainFormSpec.scala]: Can you change the invalid string to something like "a" so it is different to the previous tests please?

File: test/views/nonResident/PreviousLossOrGainViewSpec.scala [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+      "have the class 'button'" in {
+        continueButton.hasClass("button") shouldBe true
+      }
+    }
+
+    "have a sidebar that" should{
+      "have a link for help with CGT" which {
+        lazy val linkOne = document.body.getElementById("capitalGainsTaxHelp")
+
+        s"have the text ${messages.CGTlink}" in {
```
Comment (miranda-hawkes) [inline test/views/nonResident/PreviousLossOrGainViewSpec.scala]: I think this and the other link in the sidebar also need the text: (opens in new window)
(There is an example in the annualExemptAmount view). It's not in the prototype sorry but I seem to remember there being a requirement to show this whenever a link is about to take them somewhere else. Feel free to challenge me/ask someone else on that

**Pull request merged**